test/tools/test_tools_identifiers.py

```diff
@@ -166,13 +166,28 @@ def test_identifier_compare_invalid(wrong_type):
         identifier < wrong_type
 
 
-def test_identifier_is_nick():
-    assert identifiers.Identifier('Exirel').is_nick()
+@pytest.mark.parametrize('name, chantypes', (
+    ('Exirel', identifiers.DEFAULT_CHANTYPES),
+    ('@Exirel', identifiers.DEFAULT_CHANTYPES),
+    ('+Exirel', ('#',)),
+    ('!Exirel', ('#',)),
+    ('&Exirel', ('#',)),
+))
+def test_identifier_is_nick(name, chantypes):
+    assert identifiers.Identifier(name, chantypes=chantypes).is_nick()
 
 
-def test_identifier_is_nick_channel():
-    assert not identifiers.Identifier('#exirel').is_nick()
+@pytest.mark.parametrize('name, chantypes', (
+    ('#Exirel', identifiers.DEFAULT_CHANTYPES),
+    ('+Exirel', identifiers.DEFAULT_CHANTYPES),
+    ('!Exirel', identifiers.DEFAULT_CHANTYPES),
+    ('&Exirel', identifiers.DEFAULT_CHANTYPES),
+    ('@Exirel', ('#', '@')),
+))
+def test_identifier_is_nick_channel(name, chantypes):
+    assert not identifiers.Identifier(name, chantypes=chantypes).is_nick()
 
 
 def test_identifier_is_nick_empty():
     assert not identifiers.Identifier('').is_nick()
+    assert not identifiers.Identifier('', chantypes=('',)).is_nick()
```

Comment:
You know, this whole block now makes me wonder why the behavior exists. It [goes all the way back](https://github.com/sopel-irc/sopel/commit/f1182ffe6c3f2431a2058613a91a0b91b00747c6#diff-600b52f13ab78aec82ba0dedb4f3434275ba5d68b5de8f708ced1a49828f697f) to the Willie days when `Identifier` was still called `Nick`, some 8 years ago.

Even then, it was questionably useful—but now it's (at best) a dangerous pitfall, one left undocumented. One might reasonably assume that an `Identifier` for which `is_nick()` returns `False` is, therefore, a channel name… But an empty identifier represents neither.

Surely this isn't the PR in which to do it, but maybe one of us should consider making `Identifier('')` a `ValueError`. Is there ever a real-world case when an `Identifier` might be empty? I can't think of one. Channels must have names (and a hypothetical server that allowed a channel with no name would surely enforce at least the prefix character). A nickname of `''` makes no sense, since the server would reply `ERR_NONICKNAMEGIVEN` if a client tried to use it.

Reply:
I agree with you. It's time to add a `ValueError` when trying to create an `Identifier` from an empty string.

Reply:
Also, I think `Identifier` should have a `.is_channel()` method.

Reply:
Couldn't have said it better myself! Should I put that patch on my to-do list for tonight? 😁

Reply:
Alright, I tried to change that and raise an exception, however there are numerous side effects:

* When creating a PreTrigger, Sopel always set PreTrigger.nick, even if empty
* Setting PreTrigger.nick = None fix that but...
* Rule matching check if pretrigger.nick is the bot's nick, which fail if pretrigger.nick is None (could be fix by being less stupid with `Identifier.__eq__`)
* Fixing that leads to another error down the chain, and so on, and so on...

So, given the scope of this PR, I think it's best to keep that for a future PR. The test is here to check the current behavior, it'll be helpful.

Also, this will be yet another breaking change for Sopel 8, but we are used to that by now, aren't we?

Reply:
I never expected such a change to be added in this PR, haha. Adding a `.is_channel()` method will be a patch of its own (probably from me, assuming I don't forget about it in the next few days), and some kind of overhaul to make `''` an invalid `Identifier` should also be a patch of its own—probably after the simpler method addition.